code/game/objects/structures/window.dm

```diff
@@ -181,19 +181,20 @@
 
 	add_fingerprint(user)
 
-	if(istype(I, /obj/item/weldingtool) && user.a_intent == INTENT_HELP)
-		if(obj_integrity < max_integrity)
-			if(!I.tool_start_check(user, amount=0))
-				return
-
-			to_chat(user, "<span class='notice'>You begin repairing [src]...</span>")
-			if(I.use_tool(src, user, 40, volume=50))
-				obj_integrity = max_integrity
-				update_nearby_icons()
-				to_chat(user, "<span class='notice'>You repair [src].</span>")
-		else
-			to_chat(user, "<span class='warning'>[src] is already in good condition!</span>")
-		return
+	if(user.a_intent == INTENT_HELP)
+		if(istype(I, /obj/item/weldingtool) || (user.has_trait(TRAIT_HANDY) && (istype(I, /obj/item/wrench))))
```

Comment:
How does this react with the deconstruction of windows?

Reply:
Oof, good catch I can't believe I forgot to test that. Can't think of a way to fix it that isn't even more hacky so I think this is dead for now.